metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py

```diff
@@ -22,7 +22,20 @@ class DateutilTimePeriodAdjuster(TimePeriodAdjuster):
 
     def _relative_delta_for_window(self, time_granularity: TimeGranularity, count: int) -> relativedelta:
         """Relative-delta to cover time windows specified at different grains."""
-        if time_granularity is TimeGranularity.DAY:
+        if time_granularity is TimeGranularity.NANOSECOND:
+            # TODO: figure out a workaround when enabling time constraints
```

Comment:
Eh, maybe we just drop support for nanosecond for now, although I guess for cumulative metrics this isn't a big deal.

Note: BigQuery, Redshift, and Postgres only support microsecond. Trino and DuckDB support nanosecond although DuckDB uses a different type for it (and Trino requires a more explicit precision for anything smaller than millisecond).

Reply:
As discussed offline, we'll keep as is.